src/tools/chdman.cpp

```diff
@@ -1910,6 +1923,12 @@ static void do_create_hd(parameters_map &params)
 			if (err != CHDERR_NONE)
 				report_error(1, "Error adding hard disk metadata: %s", chd_file::error_string(err));
 		}
+		else if (!template_idnt_str.empty())
+		{
+			err = chd->write_metadata(HARD_DISK_IDENT_METADATA_TAG, 0, template_idnt_str);
```

Comment:
IDENT metadata is not some string with device name, but IDENTIFY_DEVICE ATA command reply, 512-byte binary structure, described for example here https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/ata/ns-ata-_identify_device_data

Reply:
So, I based this off https://wiki.mamedev.org/index.php/Driver:InterPro and https://wiki.mamedev.org/index.php/Driver:MIPS which describe using `chdman addmeta -t IDNT` to set the tag to identify the drive manufacturer/model. And I can confirm that this works, in the sense that whatever I'm running inside MAME recognizes the drive as the correct make and model, and it unblocks installing stuff like HP-UX or CLIX that can be picky about the drive. I mostly added it here as it seems a useful convenience feature, given that it's trivial to generate the string from the manufacturer/model that we already have in the template.

Reply:
Digging further, it looks like that specifically works because of how src/devices/bus/nscsi/hd.cpp is implemented: the `SC_INQUIRY` operation uses the contents of the `IDNT` tag to construct its reply. And all systems I tested this with are SCSI, which makes sense.

Reply:
agree, for SCSI drives have IDENT metadata in different format, but, for IDE drives it should not be automatically added as line of text.

Reply:
in general, I don't think it is good idea. chdman is general purpose tool, while adding device vendor/name texts with no further data looks like a hack, to make happy specific systems. this may be not enough for some other devices, and will confuse users.

Reply:
Would you be ok with gating this behind an option flag like `--genident` and making it disabled by default?

Reply:
I think this is fine without the additional flag.  If the user is selecting a template, they already have a specific target system in mind.

Reply:
@rb6502 I'm mainly saying - this PR does NOT generate SCSI INQUIRY reply. as you may see for example here https://github.com/mamedev/mame/blob/3624b416ba81e5e17f051005f5bd39cb1f6d1130/src/devices/bus/nscsi/hd.cpp#L168
vendor/device/version texts followed by vendor-specific data. proposed implementation may be considered as some platform-specific hack, for the systems which is happy with only text part, and may confuse users which expecting full valid IDENT data will be generated by chdman.
it is also incorrect and will confuse users who uses templates and chose fro example 1st from the list we have Conner CFA170A (IDE) drive.

anyway, I'll left decision if it acceptable or not to @rb6502 or @cuavas

Reply:
as of this part 
> If @p1pkin wants to add support for ATA-specific templates that populate the entire IDENTIFY_DEVICE_ATA structure that's fine too

we mainly use full IDENT reply in the case of various protections, which expecting to see some specific data in various fields, or use part of unique serial number as decryption key or data offset, etc. obviously these cant be simply generated from template, but will require platform specific quirks and hacks. and I'm not sure if chdman is right place for such things.

Reply:
> vendor/device/version texts should be followed by vendor-specific data. proposed implementation may be considered as some platform-specific hack, for the systems which is happy with only text part, and may confuse users which expecting full valid IDENT data will be generated by chdman.
> it is also incorrect and will confuse users who uses templates and chose for example 1st from the list we have Conner CFA170A (IDE) drive.

Maybe we should extend the `hd_template` struct to also include the interface? That would make it easy to handle this differently depending on whether it's ATA or SCSI or something else. And yeah, whatever we generate here is always going to be a stub, if something needs specific data that needs to be populated manually.